sections/futures/ProfitCalculator/ProfitCalculator.tsx

```diff
@@ -0,0 +1,251 @@
+					setExitPrice(ethers.BigNumber.from(clampedInput * scalar));
+				}
+
+				if (_stateVarName === 'stopLoss') {
+					const clampedInput = parseFloat(parseFloat(_stateVar).toFixed(2));
+					setStopLoss(ethers.BigNumber.from(clampedInput * scalar));
+				}
+
+				if (_stateVarName === 'marketAssetPositionSize') {
+					const clampedInput = parseFloat(parseFloat(_stateVar).toFixed(2));
+					setMarketAssetPositionSize(ethers.BigNumber.from(clampedInput * scalar));
+				}
+			}
+		}
+	};
+
+	/**
+	 * @todo Does this form need to do anything once the PositionButton is clicked?
+	 */
+	const handleCalculateProfit = (e: any) => {
+		e.preventDefault();
+	};
+
+	const setTargetInputValue = (source: string, target: string) => {
+		// We set the type of these `HTMLEement`s to `any` to ignore lint errors uwu
+		let src_: any = document.getElementById(source),
+			target_: any = document.getElementById(target);
+
+		const scalePercentage = 100;
+
+		if (src_ !== null && target_ !== null) {
+			if (src_.value !== null && target_.value !== null) {
+				if (source === 'exit-price') {
+					const gainPercent_: number = parseFloat(src_.value) / entryPrice.toNumber();
+					const clampedGainPercent = (gainPercent_ * scalePercentage).toPrecision(3);
+
+					target_.value = clampedGainPercent;
+				}
+
+				if (source === 'stop-loss') {
+					const lossPercent_: number = parseFloat(src_.value) / stopLoss.toNumber();
+					const clampedLossPercent = (lossPercent_ * scalePercentage).toPrecision(3);
+
+					target_.value = clampedLossPercent;
+				}
+
+				if (source === 'market-position-size') {
+					const basePositionSize_: number =
+						(parseFloat(src_.value) * entryPrice.toNumber()) / scalar;
+					const clampedBasePositionSize = basePositionSize_.toFixed(2);
+
+					target_.value = clampedBasePositionSize;
+				}
+			}
+		}
+	};
+
+	// Run this whenever `exitPrice` and `stopLoss` are updated
+	useEffect(() => {
+		setTargetInputValue('exit-price', 'gain-percent');
+		setTargetInputValue('stop-loss', 'loss-percent');
+		setTargetInputValue('market-position-size', 'base-position-size');
+	}, [exitPrice, stopLoss, marketAssetPositionSize]);
+
+	return (
+		<>
+			<BaseModal
+				onDismiss={() => setOpenProfitCalcModal(false)}
+				isOpen={true}
+				/**
+				 * @todo IDK how to make this comply with this project's style, e.g.
+				 *       `t('modals.confirm-transaction.title')`
+				 */
+				title={t('Profit Calculator')}
+			>
+				<ModalWindow>
+					<form onSubmit={handleCalculateProfit}>
+						<LabelWithInput
+							defaultValue={marketAssetRate}
+							labelText={'Entry Price: '}
+							placeholder={'$43,938.11'}
+							onChange={(e: any) => handleSetInput(e, entryPrice, 'entryPrice')}
+						/>
+						<ProfitCalcGrid>
+							{/* LEFT column */}
+							<LeftColumn>
+								<LabelWithInput
+									id={'exit-price'}
+									labelText={'Exit Price: '}
+									placeholder={'$46,939.11'}
```

Comment:
I'd prefer placeholders that are derived from the input. So if the user has `ETH-PERP` selected, we take the `sETH` price and add 5% for exit price, decrease by 5% for stop loss.